otel/conf.go

```diff
@@ -492,6 +523,52 @@ func validateCollectionInterval(collectionInterval string, pluginName string) (b
 	return true, nil
 }
 
+type MSSQL struct {
```

Comment:
The `CollectionInterval ` related logic among hostmetrics, iis and mssql seem identical. Can we combine them to a helper function to reduce code duplication?

Reply:
yes, exactly. Or @davidbtucker are there any solution in golang that's better than a helper function

Reply:
We could potentially use embedding, but I don't think it's worth it here yet.  Let's address it if there's more duplication in the future.

Reply:
SG